packages/server/src/workers/utils.ts

```diff
@@ -116,6 +116,28 @@ export function isDeleteInteraction(subscription: Subscription): boolean {
   return supportedInteractionExtension?.valueCode === 'delete';
 }
 
+export async function isFhirCriteriaMet(subscription: Subscription, currentResource: Resource): Promise<boolean> {
+  const criteria = getExtension(
+    subscription,
+    'https://medplum.com/fhir/StructureDefinition/fhir-path-criteria-expression'
+  );
+  if (!criteria?.valueString) {
+    return true;
+  }
+  const history = await systemRepo.readHistory(currentResource.resourceType, currentResource?.id as string);
+  const previousResource = history.entry?.[1]?.resource as Resource;
+  // If there is no previous resource, then we can't compare and have the criteria automatically pass
+  if (!previousResource) {
```

Comment:
The spec says to pass in "empty" when there is no history:

https://hl7.org/fhir/subscriptiontopic.html#resource-trigger-create

> When using fhirPathCriteria during a create, the %previous variable will be set to empty ({}). Authors should ensure that expressions can be evaluated without errors in all applicable situations.